rss-core/rss-dao/src/main/java/es/upm/fiware/rss/dao/impl/SharingReportDaoImpl.java

```diff
@@ -57,20 +69,41 @@
             }
         }
 
+        hql += " order by l.id desc";
+
         List<SharingReport> resultList;
         try {
-            List list = this.getSession().createQuery(hql).list();
-            resultList = Collections.checkedList(list, SharingReport.class);
+            Query q = this.getSession().createQuery(hql);
+
+            q.setFirstResult(Integer.max(offset, 0));  // If offset negative, set it as 0
+
+            if (size >= 0) {
+                q.setMaxResults(size);
+            }
+
+            resultList = (List<SharingReport>) q.list();
         } catch (Exception e) {
-            return null;
+            return Optional.empty();
         }
 
         if (null == resultList || resultList.isEmpty()) {
-            resultList = null;
+            return Optional.empty();
         }
 
-        return resultList;
+        return Optional.of(resultList);
+    }
+
+    @Override
+    public Optional<Boolean> setReportPay(int id, boolean pay) {
```

Comment:
It is not needed to have a specific method for updating the object. If look at other services, for those under the \@Transactional decorator, the only thing you need to do is retrieving the report object by id and then modify the object using the setter. The changes will be automatically saved. If the decorator is not being used then you have to call the update method of the dao with the modified object